gemd/entity/base_entity.py

```diff
@@ -105,14 +106,93 @@ def to_link(self,
 
         return LinkByUID(scope=scope, id=uid)
 
+    @staticmethod
```

Comment:
Because it uses ternary logic.  If there's another instance of the object being evaluated earlier in the stack (e.g., `process.ingredients[0].process`), it returns None to indicate unknown status.  That output changes once you've fully traversed an object.  I should document this behavior.